spectec/src/backend-interpreter/interpreter.ml

```diff
@@ -658,8 +658,15 @@ and create_context (name: string) (args: value list) : AlContext.mode =
   let params = params_of_algo algo in
   let body = body_of_algo algo in
 
-  if List.length args <> List.length params then
-    raise (Exception.InvalidArg ("Args number mismatch for algorithm " ^ name));
+  if List.length args <> List.length params then (
+    raise (Exception.InvalidArg (
```

Comment:
Looks good, but can we report this message using the existing infrastructure in the util/error module, such that the region is shown in the standard manner?

Reply:
Yes, it seems that a tailored wrapper function for the error handling used `no_region` by default, and therefore the region was not printed.

For this specific case, instead of using wrapper, now it's directly printed in the following standard format:
```
spec/wasm-2.0/8-reduction.watsup:201.1-207.50: interpreter error: Expected 4 arguments for the algorithm `CVTOP` but 3 arguments are given (interpreting (CVTOP I64 I64 (EXTEND S)))
```

We would need to refactor the error message for other cases as well, but perhaps in a separate commit.